hyper_search.py

```diff
@@ -0,0 +1,126 @@
+
+
+from sklearn.model_selection import train_test_split
+import numpy as np
+
+def max_depth_search(model, X, y, cycles=20, depth_list=None, verbose=False):
+    
+    if depth_list is None:
+        depth_list = [1, 3, 6, 10, None]
+
+    highest_acc = 0
+    opt_depth = 1
+
+    for depth in depth_list:
+        
+        accuracy_list = []
+        for _ in range(cycles):
+
+            X_train, X_validation, y_train, y_validation = train_test_split(X, y, train_size=0.8)
```

Comment:
all your models should have the same validation set. put this in the main file and pass the val set to each search